### DISCUSSION TITLE
Material MKDocs - collapsible text without admonition

### DISCUSSION THREAD

**Discussion by gmoskowitzQF** [squidfunk/mkdocs-material#7658] [Q&A] [answered]
I am using the following code to create a drop-down list:

`<details><summary>Maintenance</summary>`
`Exceptions<br>`
`Items<br>`
`Vendor Items<br>`
`Vendors<br>`
`</details>`

So when you click "Maintenance" it drops down a list of the items below it (Exceptions, Items, etc.).
When I preview this in VS Code it looks like a small black arrow next to "Maintenance" and the list appears under it when you click it. This is exactly what I want.
<details><summary>Maintenance</summary>
Exceptions<br>
Items<br>
Vendor Items<br>
Vendors<br>
</details>
However when I build the site it shows up as an admonition.

![image](https://github.com/user-attachments/assets/c4655c59-ee07-4d99-a832-a333c0698fe1)

How do I create the drop down list without the admonition?

**Comment by squidfunk** [comment]
Currently not possible. Material for MkDocs applies the style to `details`. We _might_ scope it to `details[class]`, but honestly, it's the first time we hear about this request and changing this might break overrides downstream, as we're increasing the specificity of the selector. For this reason, we're currently not considering this change.

**Comment by gmoskowitzQF** [comment]
Is there any other way to achieve what I want - to get a collapsible line of text without it being an admonition?

**Reply by facelessuser** [reply, marked as answer]
I assume you could just add a bit of styling to target non-admonition style details

```css
details:not([class]) {
  all:  revert !important;
}
details:not([class]) > summary {
  all:  revert !important;
}
```

<img width="791" alt="Screenshot 2024-11-01 at 7 02 12 AM" src="https://github.com/user-attachments/assets/8e4360df-95cc-4eb9-ad04-6a7e9bf35980">

**Reply by squidfunk** [reply]
Nice! Haven't thought of `revert`. Great solution!

**Comment by gmoskowitzQF** [comment]
@facelessuser 

Can you please show me where to put your "details:not([class])" text in my existing code? I'm not sure where to put it. Thanks.

This my current code:

`<details><summary>Maintenance</summary>`
`Exceptions<br>`
`Items<br>`
`Vendor Items<br>`
`Vendors<br>`
`</details>`

**Reply by kamilkrzyskow** [reply]
@gmoskowitzQF 
You'd have to follow these instructions to add additional CSS:
- https://squidfunk.github.io/mkdocs-material/customization/#additional-css

Add the CSS above to the `extra.css` file.

**Reply by gmoskowitzQF** [reply]
Thank you! That worked. Because this is done in a css file I assume it causes all my drop-downs to appear this way. In other words, if I wanted to use the default admonition style with the blue box, could I do that for one or two instances?

**Reply by kamilkrzyskow** [reply]
Every admonition is styled with a CSS class:
- https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types

So `details:not([class])` should scope the changes only to the `<details>` without a `class="note"` attribute.

You could also go the other way around and style it with another class like this `<details class="simple-details">` and instead of `details:not([class])` use `details.simple-details`, however I have not tested this other approach.

**Reply by gmoskowitzQF** [reply]
Thank you again! The method you described above is working, so now if I want a drop down without the admonition, I just add the class name to the details tag. 

However I now have a new problem. I want the items in my drop down to be hyperlinks to different topics. But markdown is not recognizing the hyperlink format (BTW this is all new to me so if I'm not describing it correctly my apologies). 

For example, my link to the How To document works outside of the drop-down, but inside the drop down it doesn't work.

![image](https://github.com/user-attachments/assets/de0f2601-c755-44c2-a02e-6bace5142eac)
![image](https://github.com/user-attachments/assets/f1513414-a741-419f-97d6-2c8cd309debd)

Can you not have a hyperlink inside a drop down?

**Reply by kamilkrzyskow** [reply]
If you search Markdown and HTML it's the first result in the docs:
- https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/?q=markdown+html
- https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#markdown-in-html

You need to enable it under the extensions:
```yaml
markdown_extensions:
  - md_in_html
```

and later for **each parent** tag of your Markdown content add the `markdown` attribute at the end, so in your case:
`<details class="simple-details" markdown>` as the `<summary>` tag doesn't have any Markdown inside.

**Comment by gmoskowitzQF** [comment]
Cannot thank you enough @kamilkrzyskow !!
